File: src/main/java/org/formation/model/Motif.java
package org.formation.model;

import java.util.ArrayList;
import java.util.List;

import javax.annotation.Generated;
import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import javax.persistence.OneToOne;
import javax.persistence.SequenceGenerator;

import org.formation.model.view.JsonViews;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import com.fasterxml.jackson.annotation.JsonView;
@SequenceGenerator(name="seq", initialValue=1)
@Entity
public class Motif {

	@Id
	@GeneratedValue(strategy = GenerationType.SEQUENCE, generator="seq")
	@JsonView(JsonViews.Common.class)
	private Integer id;
	@JsonView(JsonViews.Common.class)
	private String motif;
	@JsonView(JsonViews.Common.class)
	private Integer prix;
	@ManyToOne
	@JoinColumn(name="motif_praticien")
	@JsonView(JsonViews.Common.class)
	private Praticien praticien;
	@OneToMany(mappedBy = "motif")
	private List<Rdv> listesRdv = new ArrayList<Rdv>();

	public Integer getId() {
		return id;
	}

	public void setId(Integer id) {
		this.id = id;
	}

	public Praticien getPraticien() {
		return praticien;
	}

	public void setPraticien(Praticien praticien) {
		this.praticien = praticien;
	}

	public List<Rdv> getListesRdv() {
		return listesRdv;
	}

	public void setListesRdv(List<Rdv> listesRdv) {
		this.listesRdv = listesRdv;
	}

	public Integer getPrix() {
		return prix;
	}

	public void setPrix(Integer prix) {
		this.prix = prix;
	}

	public String getMotif() {
		return motif;
	}

	public void setMotif(String motif) {
		this.motif = motif;
	}

	@Override
	public int hashCode() {
		final int prime = 31;
		int result = 1;
		result = prime * result + ((id == null) ? 0 : id.hashCode());
		return result;
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Motif other = (Motif) obj;
		if (id == null) {
			if (other.id != null)
				return false;
		} else if (!id.equals(other.id))
			return false;
		return true;
	}
	
}

File: adoopteundoc/src/app/model/motif.ts
import {Praticien} from './praticien';
import {Rdv} from './rdv';

export class Motif {
  constructor(private _id?: number, private _motif?: string, private _prix?: number, private _praticien?: Praticien, private _rdv?: Rdv) {

  }


  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get motif(): string {
    return this._motif;
  }

  set motif(value: string) {
    this._motif = value;
  }

  get prix(): number {
    return this._prix;
  }

  set prix(value: number) {
    this._prix = value;
  }

  get praticien(): Praticien {
    return this._praticien;
  }

  set praticien(value: Praticien) {
    this._praticien = value;
  }

  get rdv(): Rdv {
    return this._rdv;
  }

  set rdv(value: Rdv) {
    this._rdv = value;
  }
}
